src/compiler/transformers/es6.ts

```diff
@@ -1115,31 +1217,100 @@ namespace ts {
         }
 
         /**
+         * Declares a `_this` variable for derived classes and for when arrow functions capture `this`.
+         *
+         * @returns The new statement offset into the `statements` array.
+         */
+        function declareOrCaptureOrReturnThisForConstructorIfNeeded(statements: Statement[], ctor: ConstructorDeclaration, hasExtendsClause: boolean, statementOffset: number) {

```

Comment:
can you move this just after the other constructor code, maybe just before `visitParameter` ?